apps/service_providers/models.py

```diff
@@ -164,6 +180,79 @@ def get_speech_service(self) -> speech_service.SpeechService:
+        if self.type == VoiceProviderType.openai_voice_engine:
+            accepted_file_types = ["mp4", "mp3"]
+            invalid_extentions = set()
+            for file in uploaded_files:
+                file_extention = file.name.split(".")[1]
+                if file_extention not in accepted_file_types:
+                    invalid_extentions.add(f".{file_extention}")
+            if invalid_extentions:
+                string = ", ".join(invalid_extentions)
+                raise UserServiceProviderConfigError(f"File extentions not supported: {string}")
+
+    @transaction.atomic()
+    def add_files(self, files):
+        self.validate_uploaded_file_names(files)
+        if self.type == VoiceProviderType.openai_voice_engine:
+            for file in files:
+                try:
+                    # TODO: Split file extention
+                    SyntheticVoice.objects.create(
+                        name=file.name,
+                        neural=True,
+                        language="",
+                        language_code="",
+                        gender="",
+                        service=SyntheticVoice.OpenAIVoiceEngine,
+                        voice_provider=self,
+                        file=file,
+                    )
+                except IntegrityError:
+                    message = f"Unable to upload '{file.name}' voice. This voice might already exist"
+                    raise UserServiceProviderConfigError(message)
+
+    def remove_file(self, file_id: int):
+        synthetic_voice = self.syntheticvoice_set.get(file_id=file_id)
+        synthetic_voice.file.delete()
+        synthetic_voice.delete()
+
+    def get_files(self):
+        """Return the files found on the synthetic voices that points to this instance"""
+        files = []
+        for voice in self.syntheticvoice_set.all():
+            # Since the File model uses a generic FK, we cannot simply do a .values_list on a VoiceProvider query,
+            # since VoiceProvider does not have a reverse relation to `File` like SyntheticVoice has
+            if voice.file:
+                files.append(voice.file)
+        return files
+
+    def remove_file_url(self):
+        return reverse(
+            "service_providers:delete_file",
+            kwargs={"team_slug": self.team.slug, "provider_type": const.VOICE, "pk": self.id, "file_id": "000"},
+        )
+
+    def add_file_url(self):
+        return reverse(
+            "service_providers:add_file",
+            kwargs={
+                "team_slug": self.team.slug,
+                "provider_type": const.VOICE,
+                "pk": self.id,
+            },
+        )
+
+    @transaction.atomic()
+    def delete(self):
+        if self.type == VoiceProviderType.openai_voice_engine:
+            files_to_delete = self.get_files()
+            [f.delete() for f in files_to_delete]
+            self.syntheticvoice_set.all().delete(audit_action=AuditAction.AUDIT)
```

Comment:
does this not happen automatically when the provider is deleted (via the cascade delete)?

Reply:
Yeah you're right, thanks for pointing this out